Python/ast_opt.c

```diff
@@ -631,6 +632,63 @@ fold_compare(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
     return 1;
 }
 
+/* bpo-44501: Pack call arguments into a constant tuple if they are
+ *            all constants. The limits are important since the optimization
+ *            shows significance after 3 or more elements, and we don't want
+ *            to create giant tuples. */
+
+#define AST_MIN_CALL_FOLD 3
+#define AST_MAX_CALL_FOLD 100
+
+static int
+fold_call(expr_ty node, PyArena *arena, _PyASTOptimizeState *state)
+{
+    assert(node->kind == Call_kind);
+
+    asdl_expr_seq *args = node->v.Call.args;
+    Py_ssize_t nargs = asdl_seq_LEN(args);
+    Py_ssize_t nkeywords = asdl_seq_LEN(node->v.Call.keywords);
+
+    if (nkeywords || !nargs) {
+        return 1;
+    }
+
+    if (nargs < AST_MIN_CALL_FOLD || nargs > AST_MAX_CALL_FOLD) {
+        return 1;
+    }
+
+    for (Py_ssize_t i = 0; i < nargs; i++) {
+        expr_ty elt = asdl_seq_GET(args, i);
+        if (elt->kind != Constant_kind) {
+            return 1;
+        }
+    }
```

Comment:
`make_const_tuple()` already preforms this check. It can be shortened:
```c
PyObject *args_tuple = make_const_tuple(args);
if (!args_tuple) {
    return PyErr_Occurred() ? 0 : 1;
}
```

Reply:
> make_const_tuple() already preforms this check.

There are multiple paths where it can return NULL, which as it can be seen in your example can be distungished with that way though I don't feel string about that change. The other places that use `make_const_tuple` are tightly coupled with `make_const`, so depending on the implementation doesn't seem reliable, but can change that if you feel that for-loop is redundant.

Reply:
Okay, I've replaced that check with a comment about our dependence to the behavior of `make_const_tuple`.